app/controllers/step_by_step_pages_controller.rb

```diff
@@ -122,4 +128,9 @@ def set_step_by_step_page
   def step_by_step_page_params
     params.require(:step_by_step_page).permit(:title, :slug, :introduction, :description)
   end
+
+  def steps_in_step_by_step_page
```

Comment:
I don't think you need this function. In `batch` you could just use: 
`@step_by_step_page.steps.each(&:request_broken_links)`